tcms/management/tests/test_admin.py

```diff
@@ -0,0 +1,37 @@
+from django.urls import reverse
+
+from tcms.tests import LoggedInTestCase
+from tcms.tests.factories import BuildFactory, ComponentFactory
+from tcms.utils.permissions import initiate_user_with_default_setups
+
+
+class TestComponentAdmin(LoggedInTestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        initiate_user_with_default_setups(cls.tester)
+        ComponentFactory()
+
+    def test_generated_database_queries(self):
+        self.client.login(  # nosec:B106:hardcoded_password_funcarg
+            username=self.tester.username, password="password"
+        )
+        with self.assertNumQueries(12):
+            response = self.client.get(reverse("admin:management_component_changelist"))
+            self.assertContains(response, "Component 0")
```

Comment:
What is "Component 0" so that you search for it ?

Reply:
The code creates a component object using `ComponentFactory` so that it can test number of queries generated. Searching for "Component 0" isn't required though. I just did that to confirm such object exists. I was running tests using `./manage.py test tcms.management.tests.test_admin`, therefore test with "Component 0" was passing locally. I believe with the whole test suite running, the "n" parameter changes. I have updated the test such that it manages all cases. 

Do you want me to remove `self.assertContains(response, "Component 0")`?